pkg/chunk/cache/memcached.go

```diff
@@ -86,6 +109,64 @@ func (c *Memcached) Fetch(ctx context.Context, keys []string) (found []string, b
 	return
 }
 
+func (c *Memcached) fetchKeysBatched(ctx context.Context, keys []string) (found []string, bufs [][]byte, missed []string, err error) {
+	type result struct {
+		found  []string
+		bufs   [][]byte
+		missed []string
+		err    error
+	}
+
+	inputCh := make(chan []string)
+	resultsCh := make(chan *result)
+	var wg sync.WaitGroup
+
+	// Initialise workers.
+	for i := 0; i < c.cfg.Parallelism; i++ {
+		wg.Add(1)
```

Comment:
The goroutine can run to competition before the next `wg.Add(1)`.  Probably not a big deal here but generally its better to do `wg.Add(c.cfg.Partallelism)` before the loop.